README.adoc

```diff
@@ -193,6 +193,48 @@ When a workflow run associated to a pull request is completed, a report is gener
 
 > image::documentation/screenshots/workflow-run-report.png[]
 
+=== Approve workflow runs
+
+This rule applies more fine-grained protections to workflow runs
+than is provided by the basic GitHub settings. If a repository
+is https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository[set up to only allow workflow runs from committers],
+the bot can automatically approve some workflows which meet a set of rules.
+
+Syntax of the `.github/quarkus-github-bot.yml` file is as follows:
+
+[source, yaml]
+----
+features: [ APPROVE_WORKFLOWS ]
+workflows:
+      rules:
+        - allow:
+            directories:
+             - ./src
+             - ./documentation
+             files:
+             - README.md
+            users:
+              minContributions: 5
+        unless:
+           directories:
+             - ./github
+           files:
+             - bad.xml
```

Comment:
Not sure how useful this specific part will be in practice. You can execute whatever you want on CI by adjusting the Java code of particular test.

Not saying you should remove it, it's there, let's keep it. But in the Quarkus repo, I think we won't be able to use it if we really want to be safe.

Reply:
I know what you mean. I was imagining people might want to restrict the `.github` folder, since that's where the workflows live, and the various levels of `pom.xml`, since changed/added dependencies is a common vector for malware. I agree someone determined enough could write something in Java without extra dependencies, but a check on `pom.xml` might increase the bar for how determined they'd have to be. The downside of the restrictions on `pom.xml` is that many non-trivial features would also change it, so restricting it would mean more manual approval work. 

It might be worth changing the example to explicitly suggest `pom.xml` (realistic) rather than `bad.xml` (clear, but useless).

Reply:
Well you can change it to `pom.xml` but really no files are safe as long as they are executed. So pom, any Java file being actually run, any workflow file and maybe others I'm not thinking about.